### DISCUSSION TITLE
Copilot neovim support plugin github/copilot.vim failed to authenticate on server

### DISCUSSION THREAD

**Discussion by submartingales** [community/community#12162] [Copilot Conversations] [answered]
I'm running neovim in TUI on a remote server through ssh. When I type `:Copilot setup`, as expected no browser will be opened up cause there is no dekstop environment installed. I suggest `github/copilot.vim` may provide a  smooth way to authenticate on a remote server.

**Comment by tpope** [comment, marked as answer]
This is accounted for. You should get a message like `Failed to open browser.  Visit https://` which will tell you the URL to open by hand.

**Reply by ovidiu-ionescu** [reply]
The code doesn't seem to work, the program gets stuck in `Press ENTER to open GitHub in your browser`. To get past this point, I edited line 728 in copilot.vim and forced the condition to be false
```
- elseif len(browser)
+ elseif 0
```
I would suggest press `Enter to open the browser, press M to do it manually`. That way the user has a choice if browser detection goes wrong.

**Reply by tpope** [reply]
The latest version features an undocumented variant `:Copilot! setup` to skip opening the browser.

**Reply by flavio87** [reply]
For me I'm actually just getting a stalled screen as soon as I type `:Copilot setup` or `:Copilot! setup`, no messages with URL to visit. Any pointers?

**Reply by tpope** [reply]
I would start by waiting a few minutes, in case an HTTP request is slowly timing out.

**Reply by sudoFerraz** [reply]
Same as @flavio87 for me, both options do not return anything;

`:Copilot` status also doesn't return nothing, I can open` :help copilot `normally tho

**Reply by p-patil** [reply]
I have the same problem. Just installed it locally with neovim, running `:Copilot setup` just responds `Copilot: 'setup'` and does nothing.

**Reply by tpope** [reply]
I screwed up an error message in the last release; please try upgrading. :pray:

**Reply by p-patil** [reply]
No problem! I actually realized my neovim version was 0.4, the default in the Ubuntu repositories, and once I upgraded to the developer version and downgraded node to 16 it worked :)
